exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -216,46 +218,58 @@ func (prwe *prwExporter) export(ctx context.Context, requests []*prompb.WriteReq
+	backOff.InitialInterval = 1 * time.Second
+	backOff.MaxInterval = 10 * time.Second
+	backOff.MaxElapsedTime = 1 * time.Minute
+
+	retryFunc := func() error {
+		// Uses proto.Marshal to convert the WriteRequest into bytes array
+		data, err := proto.Marshal(writeReq)
+		if err != nil {
+			return consumererror.NewPermanent(err)
+		}
+		buf := make([]byte, len(data), cap(data))
+		compressedData := snappy.Encode(buf, data)
 
-	// Create the HTTP POST request to send to the endpoint
-	req, err := http.NewRequestWithContext(ctx, "POST", prwe.endpointURL.String(), bytes.NewReader(compressedData))
-	if err != nil {
-		return consumererror.NewPermanent(err)
-	}
+		// Create the HTTP POST request to send to the endpoint
+		req, err := http.NewRequestWithContext(ctx, "POST", prwe.endpointURL.String(), bytes.NewReader(compressedData))
+		if err != nil {
+			return consumererror.NewPermanent(err)
+		}
 
-	// Add necessary headers specified by:
-	// https://cortexmetrics.io/docs/apis/#remote-api
-	req.Header.Add("Content-Encoding", "snappy")
-	req.Header.Set("Content-Type", "application/x-protobuf")
-	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
-	req.Header.Set("User-Agent", prwe.userAgentHeader)
+		// Add necessary headers specified by:
+		// https://cortexmetrics.io/docs/apis/#remote-api
+		req.Header.Add("Content-Encoding", "snappy")
+		req.Header.Set("Content-Type", "application/x-protobuf")
+		req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+		req.Header.Set("User-Agent", prwe.userAgentHeader)
 
-	resp, err := prwe.client.Do(req)
-	if err != nil {
-		return consumererror.NewPermanent(err)
-	}
-	defer resp.Body.Close()
+		resp, err := prwe.client.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		// 2xx status code is considered a success
+		// 5xx errors are recoverable and the exporter should retry
+		// Reference for different behavior according to status code:
+		// https://github.com/prometheus/prometheus/pull/2552/files#diff-ae8db9d16d8057358e49d694522e7186
+		if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+			return nil
+		}
 
-	// 2xx status code is considered a success
-	// 5xx errors are recoverable and the exporter should retry
-	// Reference for different behavior according to status code:
-	// https://github.com/prometheus/prometheus/pull/2552/files#diff-ae8db9d16d8057358e49d694522e7186
-	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
-		return nil
-	}
-	body, err := io.ReadAll(io.LimitReader(resp.Body, 256))
-	rerr := fmt.Errorf("remote write returned HTTP status %v; err = %w: %s", resp.Status, err, body)
-	if resp.StatusCode >= 500 && resp.StatusCode < 600 {
-		return rerr
+		body, err := io.ReadAll(io.LimitReader(resp.Body, 256))
+		rerr := fmt.Errorf("remote write returned HTTP status %v; err = %w: %s", resp.Status, err, body)
+		if resp.StatusCode >= 500 && resp.StatusCode < 600 {
+			return rerr
+		}
+		return consumererror.NewPermanent(rerr)
```

Comment:
you have to use [this permanent error.](https://pkg.go.dev/github.com/cenkalti/backoff/v4#PermanentError) and then convert to` consumererror.NewPermanent(rerr)` when getting the return value of `backoff.Retry` no?